server/src/lib.rs

```diff
@@ -110,29 +115,65 @@ pub struct AuthenticationResult {
     token: Option<String>,
 }
 
+#[derive(Debug,Clone)]
+enum AuthenticationInfos {
```

Comment:
I would probably rename it to `AuthenticationContext`, which seems more relevant with the storage of the Kerberos state.